lib/TPP/MapToBatchReduceGEMM.cpp

```diff
@@ -59,6 +85,45 @@ static bool isOutputOperand(linalg::LinalgOp linalgOp, OpOperand &operand) {
   return !isInputOperand(linalgOp, operand);
 }
 
+static LogicalResult checkVNNIAccessPatterns(linalg::LinalgOp linalgOp) {
+  SmallVector<AffineMap> maps;
+  for (OpOperand &operand : linalgOp->getOpOperands()) {
+    AffineMap map = linalgOp.getMatchingIndexingMap(&operand);
+    if (isInputOperand(linalgOp, operand)) {
+      if (map.getNumResults() < 3)
+        return failure();
+      if (operand.getOperandNumber() == 1) {
+        assert(map.getNumResults() == 5);
```

Comment:
not sure if asserting is the right think as we are checking the access pattern, bailing out for me would be better.